pandas/tests/indexes/datetimes/test_constructors.py

```diff
@@ -950,3 +950,10 @@ def test_datetimeindex_constructor_misc(self):
         )
         assert len(idx1) == len(idx2)
         assert idx1.freq == idx2.freq
+
+
+def test_timedelta_constructor_identity():
+    # Test for #30543
+    expected = pd.Timedelta(np.timedelta64(1, "s"))
+    result = pd.Timedelta(expected)
+    assert result is expected
```

Comment:
I'm not sure about this expectation - we don't do this for Timestamps so not sure should do it for Timedeltas as well but maybe @jschendel has thoughts

Reply:
There is a test for the constructor identity too in the Timestamp shortcut PR #30676. Sorry for posting the Timedelta one before that one gets merged. It's just that the discussion was mostly done there, so I thought I'd get on the Timedelta one too.